RecoParticleFlow/PFProducer/src/PFAlgo.cc

```diff
@@ -461,122 +429,30 @@ void PFAlgo::processBlock( const reco::PFBlockRef& blockref,
-    {
-      for(unsigned iEle=0; iEle<elements.size(); iEle++) {
-	PFBlockElement::Type type = elements[iEle].type();
-	if(type==PFBlockElement::TRACK)
-	  {
-	    if(elements[iEle].trackRef()->algo() == reco::TrackBase::conversionStep)
-	      active[iEle]=false;
-	    if(elements[iEle].trackRef()->quality(reco::TrackBase::highPurity))continue;
-	    const auto* trackRef = dynamic_cast<const reco::PFBlockElementTrack*>((&elements[iEle]));
-	    if(!(trackRef->trackType(reco::PFBlockElement::T_FROM_GAMMACONV)))continue;
-	    if(!elements[iEle].convRefs().empty())active[iEle]=false;
-	  }
+void PFAlgo::conversionAlgo(const edm::OwnVector<reco::PFBlockElement> &elements, std::vector<bool>& active) {
+  for(unsigned iEle=0; iEle<elements.size(); iEle++) {
+    PFBlockElement::Type type = elements[iEle].type();
+    if(type==PFBlockElement::TRACK)
+      {
+        if(elements[iEle].trackRef()->algo() == reco::TrackBase::conversionStep)
+          active[iEle]=false;
+        if(elements[iEle].trackRef()->quality(reco::TrackBase::highPurity))continue;
+        const auto* trackRef = dynamic_cast<const reco::PFBlockElementTrack*>((&elements[iEle]));
+        if(!(trackRef->trackType(reco::PFBlockElement::T_FROM_GAMMACONV)))continue;
+        if(!elements[iEle].convRefs().empty())active[iEle]=false;
       }
-    }
-
-
-
-
-
-  if(debug_)
-    cout<<endl<<"--------------- loop 1 ------------------"<<endl;
-
-  //COLINFEB16
-  // In loop 1, we loop on all elements.
-
-  // The primary goal is to deal with tracks that are:
-  // - not associated to an HCAL cluster
-  // - not identified as an electron.
-  // Those tracks should be predominantly relatively low energy charged
-  // hadrons which are not detected in the ECAL.
-
-  // The secondary goal is to prepare for the next loops
-  // - The ecal and hcal elements are sorted in separate vectors
-  // which will be used as a base for the corresponding loops.
-  // - For tracks which are connected to more than one HCAL cluster,
-  // the links between the track and the cluster are cut for all clusters
-  // but the closest one.
-  // - HF only blocks ( HFEM, HFHAD, HFEM+HFAD) are identified
-
-  // obsolete comments?
-  // loop1:
-  // - sort ecal and hcal elements in separate vectors
-  // - for tracks:
-  //       - lock closest ecal cluster
-  //       - cut link to farthest hcal cluster, if more than 1.
-
-  // vectors to store indices to ho, hcal and ecal elements
-  vector<unsigned> hcalIs;
-  vector<unsigned> hoIs;
-  vector<unsigned> ecalIs;
-  vector<unsigned> trackIs;
-  vector<unsigned> ps1Is;
-  vector<unsigned> ps2Is;
-
-  vector<unsigned> hfEmIs;
-  vector<unsigned> hfHadIs;
-
-  vector<bool> deadArea(elements.size(), false);
+  }
+}
 
+void PFAlgo::elementLoop(const reco::PFBlock &block, reco::PFBlock::LinkData& linkData, const edm::OwnVector<reco::PFBlockElement> &elements, std::vector<bool>& active, const reco::PFBlockRef &blockref, ElementIndices& inds, std::vector<bool> &deadArea) {
```

Comment:
Lines as such could be more better readable if splitted into some reasonable way.
This will be automatically done anyhow in the clang cleaning campaign ongoing